### PR TITLE
Re-enable search survey by sid, special casing pgsql db instances.

### PR DESCRIPTION
**Pull request by sickpig**

### CODE REVIEW

**Comment by LouisGac** [comment]
Thanks for the job.

**Comment by Shnoulle** [comment]
CAST can not do the job ? CAST seem more clear for me

**Comment by sickpig** [comment]
sure we can use cast() 

the '::' operator is just a shortcut

File: application/models/Survey.php [hunk 1]
```diff
@@ -746,6 +746,8 @@ public function search()
         $criteria->join .='LEFT JOIN {{users}} AS users ON ( users.uid = t.owner_id )';
         $criteria->compare('t.active', $this->active, true, 'AND');
         $criteria->compare('surveys_languagesettings.surveyls_title', $this->searched_value, true, 'AND');
+        $sid_reference = (Yii::app()->db->getDriverName() == 'pgsql' ?' t.sid::varchar' : 't.sid');
```
Comment (sickpig) [inline application/models/Survey.php]: using cast we would have: 
``
  $sid_reference = (Yii::app()->db->getDriverName() == 'pgsql' ? ' cast (t.sid as varchar) ' : 't.sid');
``

**Pull request merged**